Week3/homework/step2-2.js

```diff
@@ -1,20 +1,47 @@
 'use strict';
 
+// Test PASS
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
   // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  let i;
+  for (i = startIndex; i < stopIndex + 1; i++) {
+    numbers.push(i);
+  }
+  let ii;
+  for (ii = 0; ii < numbers.length; ii++) {
```

Comment:
3 things here:
1. It's better to declare `i` while you are writing the `for` loop rather than declaring it separately as you do right now. This is because we intend to use the variable only inside the loop for iterating over an array and not anywhere outside of it. Thus, it's better to write it as `for(let i = 0; i < arr.length; i++)`.
2. You need not write separate for loops here for creating the `numbers` array and then iterating it for callback logic. You can combine both these actions in a single for loop like
```
for (let i = startIndex; i <= stopIndex; i++){
  numbers.push(i);
  if(i % 5 === 0 & i % 3 === 0) {
    // and so on
  }
}
```
Having two loops here doesn't impact much in this case since the iteration is very small. However, imagine a situation where you have to loop over 1 or 10 million records. It would be better to run the loop only once, wouldnt' it?
3. Mathematically speaking, a number which is divisible by 3 and 5 will be divisible by 15 as well. Thus, you can also write this logic as simply `if (numbers[ii] % 15 === 0)` instead of having the `&&` condition.